google-cloud-bigtable/samples/acceptance/read_samples_test.rb

```diff
@@ -101,9 +101,95 @@
 \tconnected_cell: 1 @#{@timestamp}
 \tconnected_wifi: 1 @#{@timestamp}
 \tos_build: PQ2A.190405.004 @#{@timestamp}
+
 OUTPUT
 
-    assert_match expected, out
+    assert_equal expected, out
+  end
+
+  it "reads_rows_nil" do
+    out, _err = capture_io do
+      reads_rows_nil bigtable_instance_id, @table_id
+    end
+    expected = <<~OUTPUT
+Reading data for phone#4c410523#20190501:
+Column Family stats_summary
+\tconnected_cell: 1 @#{@timestamp}
+\tconnected_wifi: 1 @#{@timestamp}
+\tos_build: PQ2A.190405.003 @#{@timestamp}
+
+Reading data for phone#4c410523#20190502:
+Column Family stats_summary
+\tconnected_cell: 1 @#{@timestamp}
+\tconnected_wifi: 1 @#{@timestamp}
+\tos_build: PQ2A.190405.004 @#{@timestamp}
+
+Reading data for phone#4c410523#20190505:
+Column Family stats_summary
+\tconnected_cell: 0 @#{@timestamp}
+\tconnected_wifi: 1 @#{@timestamp}
+\tos_build: PQ2A.190406.000 @#{@timestamp}
+
+Reading data for phone#5c10102#20190501:
+Column Family stats_summary
+\tconnected_cell: 1 @#{@timestamp}
+\tconnected_wifi: 1 @#{@timestamp}
+\tos_build: PQ2A.190401.002 @#{@timestamp}
+
+Reading data for phone#5c10102#20190502:
+Column Family stats_summary
+\tconnected_cell: 1 @#{@timestamp}
+\tconnected_wifi: 0 @#{@timestamp}
+\tos_build: PQ2A.190406.000 @#{@timestamp}
+
+Reading data for phone001#5c10102#20190502:
+Column Family stats_summary
+\tconnected_cell: 1 @#{@timestamp}
+\tconnected_wifi: 0 @#{@timestamp}
+\tos_build: PQ2A.190406.000 @#{@timestamp}
+
+OUTPUT
+
+    assert_equal expected, out
+  end
+
+  it "reads_rows_empty" do
+    out, _err = capture_io do
+      reads_rows_empty bigtable_instance_id, @table_id
+    end
+    expected = <<~OUTPUT
```

Comment:
it is arguable whether `read_rows(keys: [])` should return empty or return all rows like `read_rows(keys: nil)`.

I thought it should return empty, but turn out all rows. (This behaviour trigger the bug in `reads_rows_sleep` below)